app/js/components/mixins/ModalEffects.react.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+
+let ModalEffects = InnerComponent => class extends React.Component {
```

Comment:
Lol! But like a function, the returned class closes over definitions in the scope where it is defined, and so the methods of the class can refer the parameters and local variables of the function.

It is as I highlighted earlier, React's new way of doing things.
Ref: https://egghead.io/lessons/react-react-fundamentals-higher-order-components-replaces-mixins
http://natpryce.com/articles/000814.html